### PR TITLE
Improve `engine.result`

### PR DESCRIPTION
**Pull request by bryanforbes**
### Description
Improve `engine.result`

### Checklist

This pull request is:

- [ ] A documentation / typographical error fix
	- Good to go, no issue or tests are needed
- [X] A short code fix
	- please include the issue number, and create an issue if none exists, which
	  must include a complete example of the issue.  one line code fixes without an
	  issue and demonstration will not be accepted.
	- Please include: `Fixes: #<issue number>` in the commit message
	- please include tests.   one line code fixes without tests will not be accepted.
- [ ] A new feature implementation
	- please include the issue number, and create an issue if none exists, which must
	  include a complete example of how the feature would look.
	- Please include: `Fixes: #<issue number>` in the commit message
	- please include tests.

**Have a nice day!**

### CODE REVIEW

File: sqlalchemy-stubs/engine/result.pyi [hunk 1]
```diff
@@ -1,114 +1,134 @@
...
+    if sys.version_info < (3, 0):
+        def next(self) -> Row: ...
+    def partitions(self, size: Optional[int] = ...) -> Iterator[List[Row]]: ...
     def fetchall(self) -> List[Row]: ...
-    def fetchone(self) -> Row: ...
-    def fetchmany(self, size: Optional[Int] = ...) -> List[Row]: ...
+    def fetchone(self) -> Optional[Row]: ...
+    def fetchmany(self, size: Optional[int] = ...) -> List[Row]: ...
     def all(self) -> List[Row]: ...
-    def first(self) -> Row: ...
```
Comment (zzzeek) [inline sqlalchemy-stubs/engine/result.pyi:62]: separately, we need to fix all the comment annotations in the main source for these

**Comment by zzzeek** [comment]
looks great.  I specifically made the ScalarResult/ MappedResult classes separate to make the typing work out more nicely, originally it was all merged together.

File: sqlalchemy-stubs/engine/result.pyi (hunk 1 above)
Comment (CaselIT) [reply sqlalchemy-stubs/engine/result.pyi:62]: should we just remove them?

Comment (bryanforbes) [reply sqlalchemy-stubs/engine/result.pyi:62]: Yeah, the comment annotations should be removed since we're using stubs now

Comment (CaselIT) [reply sqlalchemy-stubs/engine/result.pyi:62]: added an issue https://github.com/sqlalchemy/sqlalchemy/issues/6207

**Pull request merged**